### PR TITLE
Support for `GetBlockCertificates`

### PR DESCRIPTION
**Pull request by EmilLa1**
## Purpose

Introduced the endpoint and types required to support `GetBlockCertificates`.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: v2/concordium/service.proto [hunk 1]
```diff
@@ -253,4 +253,11 @@ service Queries {
   // This endpoint is only supported for protocol version 6 and onwards.
   // If the protocol does not support the endpoint then an  'IllegalArgument' error is returned.
   rpc GetBakersRewardPeriod(BlockHashInput) returns (stream BakerRewardPeriodInfo);
+
+  // For a non-genesis block, this returns the quorum certificate, a timeout
+  // certificate (if present) and epoch finalization entry (if present).
+  // If the node does not support ConcordiumBFT (protocol versions < 6), then
+  // the response will be a grpc error (invalid argument) as the 'BlockHashInput'
+  // does not point to a block produced via ConcordiumBFT.
```
Comment (abizjak) [inline v2/concordium/service.proto:261]: This does not explain the behaviour entirely.

Example 1:
Node version 6 supports protocol 6, but for example it is not enabled yet. WHat is the response?

Example 2:
Protocol 6 is in effect, but BlockHashInput points to a block at a different protocol. What is the response?

I suppose you meant to say that if the block identifier points to a block from an old protocol version then the response is invalidArgument.

File: v2/concordium/types.proto [hunk 2]
```diff
@@ -3119,3 +3121,105 @@ message BakerRewardPeriodInfo {
   // Whether the baker is a finalizer or not.
   bool is_finalizer = 6;
 }
+
+// The signature of a 'QuorumCertificate'.
+message QuorumSignature {
+  // The bytes representing the raw aggregate signature.
```
Comment (abizjak) [inline v2/concordium/types.proto:3127]: You could say here that it'll always be fixed length (is it 48 or 96 bytes?)

File: v2/concordium/types.proto [hunk 3]
```diff
@@ -3119,3 +3121,105 @@ message BakerRewardPeriodInfo {
...
+// A proof that establishes that the successor block of
+// a 'EpochFinalizationEntry' is the immediate successor of
+// the finalized block.
+message SuccessorProof {
+  // The proof represented as raw bytes.
+  bytes value = 1;
+}
+
+// The epoch finalization entry is the proof that
+// makes to protocol able to advance to a new epoch.
```
Comment (abizjak) [inline v2/concordium/types.proto:3198]: "that makes to protocol"

File: v2/concordium/types.proto [hunk 4]
```diff
@@ -3119,3 +3121,105 @@ message BakerRewardPeriodInfo {
...
+  // A proof that the successor block is an immediate
+  // successor of the finalized block.
+  SuccessorProof successor_proof = 3;
+}
+
+
+// Certificates for a block for protocols supporting
+// ConcordiumBFT.
+message BlockCertificates {
+  // The quorum certificate. Is present only if the block is
```
Comment (abizjak) [inline v2/concordium/types.proto:3216]:
```suggestion
  // The quorum certificate. Is present if and only if the block is
```

**Review by abizjak — APPROVED** [review]
Please add changes to the changelog.

File: v2/concordium/types.proto (hunk 2 above)
Comment (EmilLa1) [reply v2/concordium/types.proto:3127]: 48

**Review by td202 — APPROVED** [review]

File: v2/concordium/types.proto [hunk 5]
```diff
@@ -3119,3 +3121,107 @@ message BakerRewardPeriodInfo {
...
+  // The aggregated signature by the finalization committee that witnessed
+  // the 'round' timed out.
+  TimeoutSignature aggregate_signature = 5;
+}
+
+// A proof that establishes that the successor block of
+// a 'EpochFinalizationEntry' is the immediate successor of
+// the finalized block.
+message SuccessorProof {
+  // The proof represented as raw bytes.
```
[the remark this answers is not in the document]
Comment (EmilLa1) [reply v2/concordium/types.proto:3195]: Good addition, thanks.

**Pull request merged**